src/node/historical_queries.h

```diff
@@ -50,128 +58,296 @@ namespace ccf::historical
       {}
     };
 
-    struct Request
-    {
-      RequestStage current_stage = RequestStage::Fetching;
-      crypto::Sha256Hash entry_hash = {};
-      StorePtr store = nullptr;
-
-      // Only set when recovering ledger secrets
-      std::unique_ptr<LedgerSecretRecoveryInfo> ledger_secret_recovery_info =
-        nullptr;
-    };
-
-    // These constitute a simple LRU, where only user queries will refresh an
-    // entry's priority
-    static constexpr size_t MAX_ACTIVE_REQUESTS = 10;
-
-    using Requests = std::map<consensus::Index, Request>;
-    Requests requests;
-    std::list<consensus::Index> recent_requests;
-
-    // To trust an index, we currently need to fetch a sequence of entries
-    // around it - these aren't user requests, so we don't store them, but we do
-    // need to distinguish things-we-asked-for from junk-from-the-host
-    std::set<consensus::Index> pending_fetches;
-
-    ccf::VersionedLedgerSecret get_first_known_ledger_secret()
+    ccf::VersionedLedgerSecret get_earliest_known_ledger_secret()
     {
       if (historical_ledger_secrets->is_empty())
       {
-        return network.ledger_secrets->get_first();
+        return source_ledger_secrets->get_first();
       }
 
-      auto tx = network.tables->create_read_only_tx();
+      auto tx = source_store.create_read_only_tx();
       CCF_ASSERT_FMT(
         historical_ledger_secrets->get_latest(tx).first <
-          network.ledger_secrets->get_first().first,
+          source_ledger_secrets->get_first().first,
         "Historical ledger secrets are not older than main ledger secrets");
 
       return historical_ledger_secrets->get_first();
     }
 
-    void request_entry_at(consensus::Index idx)
+    struct StoreDetails
+    {
+      RequestStage current_stage = RequestStage::Fetching;
+      crypto::Sha256Hash entry_digest = {};
+      StorePtr store = nullptr;
+      bool is_signature = false;
+    };
+    using StoreDetailsPtr = std::shared_ptr<StoreDetails>;
+
+    struct Request
     {
-      // To avoid duplicates, remove index if it was already requested
-      recent_requests.remove(idx);
+      consensus::Index first_requested_idx = 0;
```

Comment:
Index is the Raft term, we want to use seqno in consensus contexts, and version in the KV and possibly the ledger, but I think we would like to avoid introducing new references to index as much as possible.

Reply:
I agree - I'll go through and change these in historical queries from `consensus::Index` to `kv::Consensus::SeqNo`, and rename all the `_idx` to `_seqno`. Thinking more broadly, `consensus::Index` should really be renamed to `consensus::SeqNo`, and `kv::Consensus::SeqNo` shouldn't exist (it should just be `consensus::SeqNo`. But that's a lot of renames, so I'll punt for this PR.

Reply:
Yeah, not everything needs cleaning up in the same PR, but we can at least avoid introducing more things that'll need cleaning up.

Reply:
I've now done this. This involved a lot more casting than I expected, because `kv::SeqNo` is currently signed. I think we want `SeqNo` to be unsigned and only `Version` needs to (maybe) be signed, to handle deletions/other special values? Its possible both can be unsigned with `0` or `MAX` as `NoVersion`.